db/migrate/20160307120453_remove_hostgroups_count_from_puppetclasses.rb

```diff
@@ -0,0 +1,12 @@
+class RemoveHostgroupsCountFromPuppetclasses < ActiveRecord::Migration
+  def up
+    remove_column :puppetclasses, :hostgroups_count
+  end
+
+  def down
+    add_column :puppetclasses, :hostgroups_count, :integer, :default => 0
+    HostgroupClass.all.each do |hgc|
+      Puppetclass.increment_counter(:hostgroups_count, hgc.puppetclass_id)
```

Comment:
Something along the lines of:
```sql
UPDATE puppetclasses SET hostgroups_count = (SELECT count(*) FROM hostgroup_classes WHERE hostgroup_classes.class_id = puppetclasses.id)
```
Technically a nested query, but I think most DBs should be able to optimize it quite well.